protoc-gen-swagger/genswagger/template.go

```diff
@@ -504,7 +504,7 @@ func schemaOfField(f *descriptor.Field, reg *descriptor.Registry, refs refMap) s
 		}
 	}
 
-	if j, err := extractJSONSchemaFromFieldDescriptor(fd); err == nil {
+	if j, err := extractJSONSchemaFromFieldDescriptor(f.FieldDescriptorProto); err == nil {
```

Comment:
Is this really the correct change? It looks like we're doing some sort of map parsing on line 443-445. Is that still right?

Reply:
443 - 445 is just for get `FieldDescriptorProto` of map's value, then populate `swaggerSchemaObject.AdditionalProperties`.

From map's value `FieldDescriptorProto`, we cann't extract JSONSchema(options.openapiv2_field), because it's always nil.

The JSONSchema(options.openapiv2_field) should extract from `f.FieldDescriptorProto`.